lib/metasploit/framework/login_scanner/base.rb

```diff
@@ -57,8 +57,8 @@ module Base
                     inclusion: { in: [true, false] }
 
           validates :bruteforce_speed,
-                    presence: false,
                     numericality: {
+                      allow_nil: true,
```

Comment:
@lsanchez-r7 Not following you, the conditional can be applied per validation. Like so:
```
validates :bruteforce_speed,
  unless: 'bruteforce_speed.nil?'
    numericality: {
      allow_nil: true,
      only_integer:             true,
      greater_than_or_equal_to: 0,
      less_than_or_equal_to:    5
    }
```

This would only affect this one validation, not the rest. If it's nil, it moves on, if not, it runs the numericality checks, I believe.